subprojects/logging/src/main/java/org/gradle/internal/deprecation/DeprecationLogger.java

```diff
@@ -31,17 +31,28 @@
  * The basic deprecation message structure is "Summary. DeprecationTimeline. Context. Advice. Documentation."
  *
  * The deprecateX methods in this class return a builder that guides creation of the deprecation message.
+ *
  * Summary is populated by the deprecateX methods in this class.
+ *
  * Context can be added in free text using {@link DeprecationMessageBuilder#withContext(String)}.
+ * It should provide any _instance specific_ details about the deprecated usage that would help a user identify
+ * the full nature of the use site. This is particularly important when the use site is not directly within user code.
```

Comment:
The term "use site" is unclear to me. Is this a well-known term? Do you mean "call site"?

Reply:
A call site is a type of use site. 

Some deprecated features are not “called”. For example, invalid annotation combinations. They are used, and used at a site.